### ISSUE TITLE
Stop video when tab put in background

### ISSUE THREAD

**Issue by balloob** [home-assistant/frontend#5437]
## The request

We should pause all video streams when the tab with the Home Assistant app is no longer visible. This can be achieved using the [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API).

This will prevent Home Assistant from consuming a lot of battery. 

I think that the best approach here is to make this available to all of Home Assistant by implementing this in the `partial-panel-resolver` and just disconnect the panel element. That way all `disconnectedCallback` will be called of every element, giving them the time to clean up.

## The alternatives

Use a lot of data in the background.

Related issues/PRs:
https://github.com/home-assistant/android/issues/542
https://github.com/home-assistant/android/issues/545

**Comment by bramkragten** [comment]
Should we also handle video in more-info?

**Comment by balloob** [comment]
Can we disconnect all dialogs from the DOM just as we do with the active panel?

**Comment by bramkragten** [comment]
I don't think we have a clear overview of all dialogs, except all element in `home-assistant`? But we could add something to the dialog manager I guess.

**Comment by balloob** [comment]
That will work. Note that more info dialog does not work with the dialog manager and would need it's own fix.

**Comment by zsarnett** [comment]
Just needs to be done for More info. This is done in lovelace
